sda_download/sda_download.go

```diff
@@ -153,6 +169,9 @@ func downloadFile(uri, token, filePath string) error {
 		return fmt.Errorf("failed to create directory, reason: %v", err)
 	}
 
+	if pubKeyBase64 != "" {
+		filePath += ".c4gh"
+	}
```

Comment:
I think that may cause some problem. The `filePath` may be provided with or without the `.c4gh` extension.  If I place this code at the beginning of the function, when the `filePath` is provided without the `.c4gh` extension, the encrypted file will still be downloaded, but without the `.c4gh` extension. 

With the current setup, when downloading unencrypted file, the downloaded file will not have the `.c4gh` extension, no matter if the `filePath` contains the `.c4gh` extension or not. 
And when downloading the encrypted file, the downloaded file will be guaranteed to have the `.c4gh` extension.

Reply:
I partially agree on that because the user needs to follow some rules and since the file is encrypted in our archive it should be given with the `.c4gh` extension. I mean this is a general comment but I am fine if you prefer it that way. Good job